libraries/apollo-gradle-plugin/rules.pro

```diff
@@ -19,12 +19,31 @@
     public static **[] values();
 }
 
-# Moshi uses reflection in StandardJsonAdapters
--keepclassmembers class com.apollographql.apollo3.ast.introspection.IntrospectionSchema$Schema$Kind extends java.lang.Enum {
-    <fields>;
+# Begin kotlinx-serialization rules - Introspection classes in apollo-ast use it for Json
+# Keep `Companion` object fields of serializable classes.
+# This avoids serializer lookup through `getDeclaredClasses` as done for named companion objects.
+-if @kotlinx.serialization.Serializable class **
+-keepclassmembers class <1> {
+    static <1>$Companion Companion;
 }
-# Keep all the introspection stuff as it uses moshi for Json
--keep class com.apollographql.apollo3.ast.introspection.** { *; }
+# Keep `serializer()` on companion objects (both default and named) of serializable classes.
+-if @kotlinx.serialization.Serializable class ** {
+    static **$* *;
+}
+-keepclassmembers class <2>$<3> {
+    kotlinx.serialization.KSerializer serializer(...);
+}
+# Keep `INSTANCE.serializer()` of serializable objects.
+-if @kotlinx.serialization.Serializable class ** {
+    public static ** INSTANCE;
+}
+-keepclassmembers class <1> {
+    public static <1> INSTANCE;
+    kotlinx.serialization.KSerializer serializer(...);
+}
+# @Serializable and @Polymorphic are used at runtime for polymorphic serialization.
+-keepattributes RuntimeVisibleAnnotations,AnnotationDefault
+# End kotlinx-serialization rules
```

Comment:
I'm curious if you can use explicit serializers to remove most (or all?) the reflection? See also https://github.com/Kotlin/kotlinx.serialization/issues/1348

Reply:
Yes, referencing `.serializer()` explicitly works! Good catch 👍
I initially had the impression that calling the generic method was equivalent since it calls `.serializer()`, but it's actually a different one